src/server.h

```diff
@@ -481,6 +501,18 @@ typedef enum {
     REPL_STATE_CONNECTED, /* Connected to primary */
 } repl_state;
 
+/* Replica rdb-connection replication state. Used in server.repl_rdb_conn_state for 
+ * replicas to remember what to do next. */
+typedef enum {
+    REPL_RDB_CONN_STATE_NONE = 0,            /* No active replication */
+    REPL_RDB_CONN_SEND_HANDSHAKE,            /* Send handshake sequence to primary */
+    REPL_RDB_CONN_RECEIVE_AUTH_REPLY,        /* Wait for AUTH reply */
+    REPL_RDB_CONN_RECEIVE_REPLCONF_REPLY,    /* Wait for REPLCONF reply */
+    REPL_RDB_CONN_RECEIVE_ENDOFF,            /* Wait for $ENDOFF reply */
+    REPL_RDB_CONN_RDB_LOAD,                  /* Loading rdb using rdb connection */
+    REPL_RDB_CONN_RDB_LOADED,
+} repl_rdb_conn_state;
```

Comment:
Can we consider `fullsync`? I feel it is clearer. Ditto for other 'RDB' references in the PR
```suggestion
} repl_fullsync_conn_state;
```

Reply:
If our intention is to make this PR independent of the specific data structure used to store the database, then I agree that `repl_fullsync_conn_state` would be a more appropriate name.

Reply:
I think `rdb` is fine for the states. If we can consolidate on 

1. feature name being "dual-connection-sync" (or "dual-connection-fullsync")
2. states being "rdb*"

I think it will be great. thoughts?

Reply:
Sounds good to me